File: app/src/main/java/com/recursia/popularmovies/data/mappers/GenreDatabaseModelToEntityMapper.java
package com.recursia.popularmovies.data.mappers;

import com.recursia.popularmovies.data.models.GenreDatabaseModel;
import com.recursia.popularmovies.domain.models.Genre;
import com.recursia.popularmovies.utils.Mapper;

public class GenreDatabaseModelToEntityMapper extends Mapper<GenreDatabaseModel, Genre> {
    @Override
    public Genre transform(GenreDatabaseModel genreDatabaseModel) {
        Genre genre = new Genre();
        genre.setId(genreDatabaseModel.getId());
        genre.setMovieId(genreDatabaseModel.getMovieId());
        genre.setName(genreDatabaseModel.getName());
        return genre;
    }
}

File: app/src/main/java/com/recursia/popularmovies/data/mappers/GenreDatabaseModelToEntityMapper.kt
package com.recursia.popularmovies.data.mappers

import com.recursia.popularmovies.data.models.GenreDatabaseModel
import com.recursia.popularmovies.domain.models.Genre
import com.recursia.popularmovies.utils.Mapper

class GenreDatabaseModelToEntityMapper : Mapper<GenreDatabaseModel, Genre>() {
    override fun transform(genreDatabaseModel: GenreDatabaseModel): Genre {
        val genre = Genre()
        genre.id = genreDatabaseModel.id
        genre.movieId = genreDatabaseModel.movieId
        genre.name = genreDatabaseModel.name
        return genre
    }
}
